### PR TITLE
APP-829 fix deploy issue

### PR DESCRIPTION
**Pull request by Rekard0**
## Description

Please include a summary of the change and be sure you follow the contributions rules we do provide [here](./CONTRIBUTIONS.md)

Task: [ID](https://aragonassociation.atlassian.net/browse/APP-829)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [x] I have updated the ``CHANGELOG.md`` file in the root folder.
- [x] I have tested my code on the test network.

### CODE REVIEW

File: packages/contracts/contracts/core/permission/IPermissionOracle.sol [hunk 1]
```diff
@@ -12,10 +12,10 @@ interface IPermissionOracle {
     /// @param _permissionId The permission identifier.
     /// @param _data Optional data passed to the `PermissionOracle` implementation.
     /// @return allowed Returns true if the call is permitted.
-    function hasPermissions(
+    function hasPermission(
```
Comment (Rekard0) [inline packages/contracts/contracts/core/permission/IPermissionOracle.sol:15]: Should we rename this to `isPermitted(...)` as described but it's notice ?

Comment (brickpop) [reply packages/contracts/contracts/core/permission/IPermissionOracle.sol:15]: To me it makes much less of a difference (hasPermission == isPermitted). But if we have a complete list of all the scopes, maybe it's easier to see?
- DAO => hasPermission(), canUse(), canPerform()
- PermissionManager => isGranted()
- PermissionOracle => isPermitted(), isAuthorized(), ...? Even isGranted()?

If the oracle is conceptually an extension of the PermissionManager, maybe it makes sense to use `isGranted()` in both?

Comment (Rekard0) [reply packages/contracts/contracts/core/permission/IPermissionOracle.sol]: amended

**Review by brickpop — DISMISSED** [review]

File: .github/helpers/contracts/dummy-dao/vote.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  if (isERC20Voting === 'erc20') {
+    VotingContract = new ethers.Contract(
+      votingAddress,
+      Erc20VotingJson.abi,
+      signer
+    );
+  } else {
+    VotingContract = new ethers.Contract(
+      votingAddress,
+      WhiteVotingJson.abi,
```
Comment (Michael-A-Heuer) [inline .github/helpers/contracts/dummy-dao/vote.js:44]: Rename to
```suggestion
      AllowVotingJson.abi,
```
and everywhere else where applicable

File: packages/contracts/CHANGELOG.md [hunk 3]
```diff
@@ -24,6 +25,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Changed
 
+- Updated deploy script to correctly use `ERC1967Proxy`.
+- Renamed `hasPermission` to `isGranted` in both `PermissionManager` and `IPermissionOracle`
```
Comment (Michael-A-Heuer) [inline packages/contracts/CHANGELOG.md:29]:
```suggestion
- Renamed `hasPermission` to `isGranted` in both `PermissionManager` and `IPermissionOracle`.
```

File: packages/contracts/contracts/core/permission/PermissionManager.sol [hunk 4]
```diff
@@ -294,18 +294,18 @@ contract PermissionManager is Initializable {
         emit Frozen(_permissionId, msg.sender, _where);
     }
 
-    /// @notice Checks if a caller has the permissions on a contract via a permission identifier and redirects the approval to an `PermissionOracle` if this was in the setup.
+    /// @notice Checks if a caller has the permissions granted on a contract via a permission identifier and redirects the approval to an `PermissionOracle` if this was in the setup.
```
Comment (Michael-A-Heuer) [inline packages/contracts/contracts/core/permission/PermissionManager.sol:297]:
```suggestion
    /// @notice Checks if a caller is granted permissions on a contract via a permission identifier and redirects the approval to an `PermissionOracle` if this was specified in the setup.
```

File: packages/contracts/test/core/permission/permission-manager.ts [hunk 5]
```diff
@@ -642,62 +642,62 @@ describe('Core: PermissionManager', function () {
     });
   });
 
-  describe('hasPermissions', () => {
+  describe('isGranted', () => {
     it('should return true fcr granted user', async () => {
```
Comment (Michael-A-Heuer) [inline packages/contracts/test/core/permission/permission-manager.ts:646]:
```suggestion
    it('should return true if the permission is granted to the user', async () => {
```
Maybe this is an opportunity to improve the other test descriptions that are poorly written.

File: packages/contracts/test/core/permission/permission-manager.ts [hunk 6]
```diff
@@ -642,62 +642,62 @@ describe('Core: PermissionManager', function () {
...
         pm.address,
         otherSigner.address,
         ADMIN_PERMISSION_ID,
         []
       );
-      expect(hasPermissions).to.be.equal(true);
+      expect(isGranted).to.be.equal(true);
     });
 
     it('should return false for non granted user', async () => {
```
Comment (Michael-A-Heuer) [inline packages/contracts/test/core/permission/permission-manager.ts:657]:
```suggestion
    it('should return false if the permissions is not granted to the user', async () => {
```

File: packages/contracts/test/core/permission/permission-manager.ts [hunk 7]
```diff
@@ -642,62 +642,62 @@ describe('Core: PermissionManager', function () {
...
         pm.address,
         otherSigner.address,
         ADMIN_PERMISSION_ID,
         []
       );
-      expect(hasPermissions).to.be.equal(false);
+      expect(isGranted).to.be.equal(false);
     });
 
     it('should return true for any who granted permission', async () => {
```
Comment (Michael-A-Heuer) [inline packages/contracts/test/core/permission/permission-manager.ts:667]:
```suggestion
    it('should return true for permissions granted to any address on a specific target contract using the `ANY_ADDR` flag', async () => {
```

File: packages/contracts/test/core/permission/permission-manager.ts [hunk 8]
```diff
@@ -642,62 +642,62 @@ describe('Core: PermissionManager', function () {
...
         pm.address,
         otherSigner.address,
         ADMIN_PERMISSION_ID,
         []
       );
-      expect(hasPermissions).to.be.equal(true);
+      expect(isGranted).to.be.equal(true);
     });
 
     it('should return true for any where granted permission', async () => {
```
Comment (Michael-A-Heuer) [inline packages/contracts/test/core/permission/permission-manager.ts:679]:
```suggestion
    it('should return true for permissions granted to a specifc address on any target contract using the `ANY_ADDR` flag', async () => {
```

**Review by Michael-A-Heuer — COMMENTED** [review]
A couple of comments, mostly NatSpec and naming suggestions.

File: .github/helpers/contracts/dummy-dao/proposal.js [hunk 9]
```diff
@@ -95,12 +95,12 @@ async function proposal() {
...
   }
 
-  let proposalTx = await VotingContract.newVote(
+  let proposalTx = await VotingContract.createVote(
     metadata,
     actions,
     0,
     0,
-    true,
+    false,
```
[the remark this answers is not in the document]
Comment (Rekard0) [reply .github/helpers/contracts/dummy-dao/proposal.js:103]: it is for execute if minimum reached, it is set to false as now i have vote.js

**Review by Michael-A-Heuer — APPROVED** [review]

**Review by novaknole — APPROVED** [review]

**Pull request merged**